OutOfSchool/OutOfSchool.WebApi/Controllers/V1/PictureController.cs

```diff
@@ -0,0 +1,323 @@
+                }
+
+                return Ok();
+            }
+            catch (ArgumentException ex)
+            {
+                return BadRequest(ex);
+            }
+            catch (Exception)
+            {
+                return BadRequest("An error occurred while uploading the picture.");
+            }
+        }
+
+        /// <summary>
+        /// Uploads Provider picture to data storage.
+        /// </summary>
+        /// <param name="providerId">Id of the Provider Entity.</param>
+        /// <param name="picture">Picture.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        [Authorize(Roles = "provider")]
+        [Authorize(Roles = "admin")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [HttpPost]
+        [Route("upload/provider/{providerId}")]
+        public async Task<IActionResult> UploadPhotoProvider(long providerId, IFormFile picture)
+        {
+            try
+            {
+                if (providerId <= 0)
+                {
+                    return BadRequest("ProviderId is invalid.");
+                }
+
+                if (picture is null)
+                {
+                    return BadRequest("Picture is missing.");
+                }
+
+                if (!IsSizeValid(picture))
+                {
+                    return BadRequest("The size of the picture invalid.");
+                }
+
+                if (!IsExtensionValid(picture))
+                {
+                    return BadRequest("Extension of the picture invalid.");
+                }
+
+                using (var fileStream = new MemoryStream())
+                {
+                    fileStream.Position = uint.MinValue;
+                    await picture.CopyToAsync(fileStream, CancellationToken.None).ConfigureAwait(false);
+                    await pictureService.UploadPictureProvider(providerId, fileStream, picture.ContentType).ConfigureAwait(false);
+                }
+
+                return Ok();
+            }
+            catch (ArgumentException ex)
+            {
+                return BadRequest(ex);
+            }
+            catch (Exception)
+            {
+                return BadRequest("An error occurred while uploading the picture.");
+            }
+        }
+
+        private bool IsSizeValid(IFormFile photo)
+        {
+            return photo.Length > maxPictureSize ? false : true;
+        }
+
+        private bool IsExtensionValid(IFormFile picture)
+        {
+            var contentType = GetFormat(picture.FileName);
+
+            return string.IsNullOrEmpty(contentType) ? false : true;
+        }
+
+        private string GetFormat(string fileName)
+        {
+            string contentType;
```

Comment:
Variable (`contentType`) declaration should be as closer to it's 1st usage as possible, ideally in same line (declaration + initialization, like `var something = specificValue;`)

It's preferred to use `var` in variable declaration to enforce beforementioned rule